mozdownload/scraper.py

```diff
@@ -892,7 +891,13 @@ def get_build_info_for_index(self, build_index=None):
 
         # If no index has been given, set it to the last build of the day.
         if build_index is None:
-            build_index = len(parser.entries) - 1
+            # Find the most recent non-empty entry.
+            build_index = len(parser.entries)
+            for build in reversed(parser.entries):
```

Comment:
No worries, these are good questions and mostly related to personal style.

As to why I'm using `reversed`: we only care about the entries at the end (the latest entries), so we want to start with those first. So rather than processing over 1000 entries, hopefully we just process 1.

As to why not enumerate: `reversed` takes a sequence and returns an iterator and `enumerate` actually returns an enumerator (not a sequence). So you'd have to convert the enumerator back to a list. Something like:
    `for build_index, build in reversed(list(enumerate(parser.entries))):` 
Which seems both unintuitive and wasteful for a list of > 1000 entries. Of course I could be convinced otherwise!

Thanks for the review.